Add … push --sign-by-sigstore

mtrmac opened the pull request:
`podman push` and `podman manifest push sync` now support `--sign-by-sigstore=param-file`,
using the `containers-sigstore-signing-params.yaml(5)` file format.

That notably adds support for Fulcio and Rekor signing.

Depends on <s>unmerged</s> https://github.com/containers/image/pull/1787 ; see that PR for documentation of the YAML file format, as well as example files.

See also https://github.com/containers/common/pull/1288 for more discussion about where the interactive prompting pieces should be.

<s>Untested so far.</s>

#### Does this PR introduce a user-facing change?

```release-note
`podman push` and `podman manifest push` now support a `--sign-by-sigstore` option, which allows using Fulcio and Rekor.
```

mtrmac commented on cmd/podman/common/sign.go:
```diff
@@ -26,9 +27,26 @@ func PrepareSigningPassphrase(pushOpts *entities.ImagePushOptions, signPassphras
+			promptForPassphrase = true
+		}
+		if pushOpts.SignBySigstoreParamFile != "" {
+			// Oops, we need to pre-parse the file because the actual copy implementation is too far from this
```
This is _ugly_, and it doesn’t help for Fulcio. Compare https://github.com/containers/common/pull/1288 .

I guess all of creating a `Signer` should happen somewhere around here already?

mtrmac commented:
Reworked, together with the c/common part; now the top-level CLI fully creates a `signer.Signer`, handling all the interactivity upfront.

This is now ready for review, apart from relying on unmerged dependencies.

vrothberg reviewed (commented):
LGTM

TomSweeneyRedHat reviewed (commented):
LGTM
A blog would be most welcomed for this functionality once it goes live.

mtrmac commented:
Now ready for review and possible merging.

vrothberg reviewed (approved):
/lgtm
/hold

rhatdan commented:
/lgtm

mtrmac commented:
Rebased, tests pass. Please merge.

(The changes since the last approved version are in `go.mod`, and an update in test mechanisms; no changes in non-test code.)

ashley-cui reviewed (commented):
/lgtm

mtrmac commented:
/hold cancel
